### PR TITLE
Fixed 500 Page Template Issues

### PR DESCRIPTION
**Pull request by clintonb**
- Restored settings_value template tag
- Updated debug error views to use the actual Django views
- Updated acceptance tests

@rocha @dsjen

### CODE REVIEW

File: analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
 
 
+@register.simple_tag
+def settings_value(name):
+    """
+    Retrieve a value from settings.
+
+    If setting is not found, None is returned.
+    """
+    return getattr(settings, name, None)
```
Comment (rocha) [inline analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py]: Maybe empty string instead of `None` as the default valu,  given that it is going to be rendered in a template?

File: acceptance_tests/test_error_pages.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from bok_choy.web_app_test import WebAppTest
+from acceptance_tests import PLATFORM_NAME, APPLICATION_NAME, SUPPORT_URL
+from pages import ServerErrorPage, NotFoundErrorPage, AccessDeniedErrorPage, AuthErrorPage
+
+
+class ErrorPagesTests(WebAppTest):
+    error_page_classes = [ServerErrorPage, NotFoundErrorPage, AccessDeniedErrorPage, AuthErrorPage]
```
Comment (dsjen) [inline acceptance_tests/test_error_pages.py:7]: The `AccessDeniedErrorPage` also has a "Logout then login" button that could be tested too.

File: analytics_dashboard/analytics_dashboard/urls.py [hunk 3]
```diff
@@ -29,7 +30,7 @@
     url(r'^accounts/logout/$', views.logout, name='logout'),
     url(r'^accounts/logout_then_login/$', views.logout_then_login, name='logout_then_login'),
     url(r'^test/auto_auth/$', views.AutoAuth.as_view(), name='auto_auth'),
-    url(r'^auth/error/$', views.AuthError.as_view(), name='auth_error'),
+    url(r'^auth/error/$', 'django.views.defaults.server_error', {'template_name': 'auth_error.html'}, name='auth_error'),
```
Comment (dsjen) [inline analytics_dashboard/analytics_dashboard/urls.py:33]: Why the change from class based views?

Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/urls.py:33]: I want to use the actual view that will be rendered in production. Also, it didn't make sense for us to have our own view when Django provides one.

File: analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py (hunk 1 above)
Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py]: This is restoring old code that was removed earlier. The reason for `None` is so that the issue is, hopefully, found earlier in development. An empty string would hide the problem if a developer supplied an incorrect setting name.

File: acceptance_tests/test_error_pages.py (hunk 2 above)
Comment (clintonb) [inline acceptance_tests/test_error_pages.py:7]: For now, I only want to test the behavior common to all pages.

File: analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py (hunk 1 above)
Comment (rocha) [inline analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py]: Seems to me that we are better raising errors or warnings if certain settings are not set.

Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py]: Updated.

Comment (rocha) [inline analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py]: Updated to equivalent code?

Comment (rocha) [inline analytics_dashboard/analytics_dashboard/templatetags/dashboard_extras.py]: Nevermind, I forgot that `getattr` raises an error if the attribute is missing.

**Comment by rocha** [comment]
:rocket:

File: analytics_dashboard/analytics_dashboard/urls.py (hunk 3 above)
Comment (dsjen) [inline analytics_dashboard/analytics_dashboard/urls.py:33]: Ok, thanks.

File: acceptance_tests/test_error_pages.py (hunk 2 above)
Comment (dsjen) [inline acceptance_tests/test_error_pages.py:7]: It seems like the exception would be the most interesting/important to test.
